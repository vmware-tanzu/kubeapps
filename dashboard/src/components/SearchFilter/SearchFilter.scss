// Copyright 2018-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

#search {
  margin-left: 0.3rem;
  color: var(--cds-global-typography-color-500, #000);
}

#search input::placeholder {
  color: var(--cds-global-typography-color-300, #454545);
}

header .search-box form {
  border-radius: 0.15rem;
  margin-left: 1rem;
}
