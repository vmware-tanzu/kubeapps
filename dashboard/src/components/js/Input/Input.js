// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import PropTypes from "prop-types";
import React from "react";

const Input = ({ children, ...fieldProps }) => (
  <div className="clr-input-wrapper">
    <input className="clr-input" {...fieldProps} />
    {children}
  </div>
);

Input.propTypes = {
  // Optional item to add. It could be a help section or an error message
  children: PropTypes.node,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  type: PropTypes.string,
};

Input.defaultProps = {
  children: null,
};

export default Input;
