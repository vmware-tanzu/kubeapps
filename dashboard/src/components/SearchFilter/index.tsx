// Copyright 2018-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import SearchFilter from "./SearchFilter";

export default SearchFilter;
