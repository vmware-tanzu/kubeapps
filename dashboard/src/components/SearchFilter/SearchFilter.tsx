// Copyright 2018-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { CdsIcon } from "@cds/react/icon";
import Column from "components/js/Column";
import Row from "components/js/Row";
import React from "react";
import Input from "../js/Input";
import "./SearchFilter.css";

export interface ISearchFilterProps {
  value: string;
  className?: string;
  placeholder: string;
  onChange: (filter: string) => void;
  submitFilters: () => void;
}

function SearchFilter(props: ISearchFilterProps) {
  const [value, setValue] = React.useState(props.value);
  const [timeout, setTimeoutState] = React.useState({} as NodeJS.Timeout);
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    // Copy is needed to avoid losing the reference
    const valueCopy = e.currentTarget.value;
    setValue(e.currentTarget.value);
    // Gather changes before submitting
    clearTimeout(timeout);
    setTimeoutState(
      setTimeout(() => {
        props.onChange(valueCopy);
      }, 300),
    );
  };
  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    props.submitFilters();
  };
  return (
    <div className="search-box">
      <form onSubmit={handleSubmit} role="search" aria-label="Search on the site">
        <Row>
          <Column span={1}>
            <CdsIcon size="sm" shape="search" />
          </Column>
          <Column span={10}>
            <Input
              id="search"
              name="search"
              type="text"
              placeholder={props.placeholder}
              autoComplete="off"
              onChange={handleChange}
              value={value}
              {...Input.defaultProps}
            />
          </Column>
        </Row>
      </form>
    </div>
  );
}

export default SearchFilter;
