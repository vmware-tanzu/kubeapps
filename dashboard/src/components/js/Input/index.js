// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import Input from "./Input";
export default Input;
