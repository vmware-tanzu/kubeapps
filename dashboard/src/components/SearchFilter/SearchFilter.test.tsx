// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { shallow } from "enzyme";
import Input from "../js/Input";
import SearchFilter, { ISearchFilterProps } from "./SearchFilter";

const defaultProps: ISearchFilterProps = {
  value: "",
  placeholder: "search!",
  onChange: jest.fn(),
  submitFilters: jest.fn(),
};

jest.useFakeTimers();

it("should render a PageHeader", () => {
  const wrapper = shallow(<SearchFilter {...defaultProps} value="test" />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(Input).prop("value")).toBe("test");
});

it("changes the filter", () => {
  const onChange = jest.fn();
  const wrapper = shallow(<SearchFilter {...defaultProps} value="test" onChange={onChange} />);
  wrapper.find(Input).simulate("change", { currentTarget: { value: "foo" } });
  jest.runAllTimers();
  expect(onChange).toHaveBeenCalledWith("foo");
});

it("should render a PageHeader (onSubmit)", () => {
  const onSubmit = jest.fn();
  const wrapper = shallow(<SearchFilter {...defaultProps} value="test" submitFilters={onSubmit} />);
  wrapper.find("form").simulate("submit", { preventDefault: jest.fn() });
  expect(onSubmit).toHaveBeenCalled();
});
