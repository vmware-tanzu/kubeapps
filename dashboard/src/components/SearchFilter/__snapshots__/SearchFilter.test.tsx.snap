// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a PageHeader 1`] = `
<div
  className="search-box"
>
  <form
    aria-label="Search on the site"
    onSubmit={[Function]}
    role="search"
  >
    <Row
      list={false}
    >
      <Column
        listItem={false}
        span={1}
      >
        <CdsIcon
          shape="search"
          size="sm"
        />
      </Column>
      <Column
        listItem={false}
        span={10}
      >
        <Input
          autoComplete="off"
          id="search"
          name="search"
          onChange={[Function]}
          placeholder="search!"
          type="text"
          value="test"
        />
      </Column>
    </Row>
  </form>
</div>
`;
